src/wrapper/objects/jmap.rs

```diff
@@ -171,6 +171,12 @@ impl<'a: 'b, 'b> JMap<'a, 'b> {
     }
 }
 
+impl Drop for JMap<'_, '_> {
+    fn drop(&mut self) {
+        let _ = self.env.delete_local_ref(self.class.into());
```

Comment:
I guess it could work, but I think we'd have to copy the object pointer. So, we'd store in `JMap` something like a `class_obj: AutoLocal`, but every time we need to call a method we have to do `let class = JClass::from(class_obj);`.

Maybe there's a simpler way?

Reply:
There is an implementation of `Desc` trait for `JClass` for `AutoLocal` that casts that AutoLocal to `JClass`. It can be used like that:

```rust
let class = env.find_class("java/util/Map")?;
let class = env.auto_local(class.into());

let get_id = env.get_method_id(&class, /* <-- It will invoke Desc#lookup casting AutoLocal to JClass */
    "get", "(Ljava/lang/Object;)Ljava/lang/Object;")?;
```

Reply:
The Desc impl:

https://github.com/jni-rs/jni-rs/blob/e89a40f45aeb50b6756e16e48cf85477f0a06c14/src/wrapper/descriptors/class_desc.rs#L34-L41

Reply:
Cool! This is very useful. I removed the `Drop` implementation and used an `AutoLocal` field in the struct. I had to use the `'b` lifetime though, because `AutoLocal` keeps a borrow of `env`.

I could also change `AutoLocal` so that it also has two lifetimes for `&'b JNIEnv<'a>`.

Reply:
> I could also change AutoLocal so that it also has two lifetimes for &'b JNIEnv<'a>.

If it will make `AutoLocal` more flexible or easier to use, would be wonderful!

*CC @DarkEld3r  @vitvakatu*

Reply:
I included the change in a separate commit. I can revert it if necessary.